extensions/wikia/Chat2/js/ChatEntryPoint.js

```diff
@@ -17,153 +17,107 @@ var ChatEntryPoint = {
+		$('.ChatModuleUninitialized').each(function () {
 			ChatEntryPoint.processModuleTemplate($(this));
 			$(this).removeClass('ChatModuleUninitialized');
 		});
 	},
 
-	// fill-in the whole module template
-	processModuleTemplate: function($t) {
-		// @todo - right now it's a custom html-based template, all the logic for inserting variables is here
-		// once the mustache is loaded on every page, rewrite the template and remove most of the code below
-		var items = [], i, cnt = window.wgWikiaChatUsers.length, img = window.wgWikiaChatProfileAvatarUrl;
-		$t.find('.chat-contents').
-			addClass((cnt) ? 'chat-room-active' :  'chat-room-empty').
-			addClass((window.wgUserName) ? 'chat-user-logged-in' :  'chat-user-anonymous');
-		$t.find('.chat-total').html(cnt);
-		$t.find('p.chat-name').html(window.wgSiteName);
-		if (!window.wgUserName) {
-			$t.find('div.chat-join button').addClass('loginToChat');
-		}
-		// we use attr instead of data because we want it to be present in dom (needed for css selector)
-		$t.find('div.chat-join button').attr('data-msg-id', cnt ? 'chat-join-the-chat' : 'chat-start-a-chat');
-
-		if (cnt) {
-			var $carousel = $t.find('ul.carousel');
-			var $u = $carousel.find('>li');
-			for( i=0 ; i < cnt ; i++ ) {
-				items.push(ChatEntryPoint.fillUserTemplate($u.clone(),window.wgWikiaChatUsers[i]));
-			}
-			$carousel.html(items);
-			ChatEntryPoint.initCarousel($t.find('.chat-whos-here'));
-		} else {
-			$t.find('ul.carousel').remove();
-		}
-		if (img) {
-			$t.find('.carousel-container img.avatar.currentUser').attr('src', img);
-		} else {
-			$t.find('.carousel-container img.avatar.currentUser').remove();
-		}
+	/**
+	 * Creates carousel of users and fills in some of fields with translated messages
+	 * @param $t chat module element
+	 */
+	processModuleTemplate: function ($t) {
+		ChatEntryPoint.initCarousel($t.find('.chat-whos-here'));
+
 		// process i18n the messages
-		$t.find('[data-msg-id]').each(function() {
+		$t.find('[data-msg-id]').each(function () {
 			var $e = $(this);
 			$e.text($.msg($e.data('msg-id'), $e.data('msg-param')));
 		});
 	},
 
-	// based on the template and user information, return a filled-in element
-	fillUserTemplate: function($t, user) {
-		if (user.showSince) {
-			var months = window.wgWikiaChatMonts || window.wgMonthNamesShort;
-			user.since = months[user.since_month] + ' ' + user.since_year;
-		}
-		$t.find('[data-user-prop]').each(function() {
-			var $e = $(this), attr = $e.data('user-attr'), p = $e.data('user-prop'), v = user[p];
-			if (typeof v === "undefined") {
-				$e.remove();
-			} else
-			if (attr) {
-				$e.attr(attr, v);
-			} else {
-				$e.html(v);
-			}
-		});
-		return $t;
-	},
+	/**
+	 * change the user list into the carousel
+	 * @param $el chat who is here element
+	 */
+	initCarousel: function ($el) {
+		var popoverTimeout = 0,
+			// differ number of users on chat according to it's width
+			isWideChat = $el.width() > 260;
```

Comment:
this may change, but carousel is already initialized and won't impact the (in this case) number of avatars in it. Basically, the `.carousel` function doesn't allow to change it's item number after initialization